consensus/src/quorum_store/utils.rs

```diff
@@ -318,8 +322,7 @@ impl ProofQueue {
                     },
                     None => {}, // Proof was already committed, skip.
                 }
-            }
-            if *expiration < current_time && !excluded_proofs.contains(digest) {
+            } else if let Some(Some(_)) = self.digest_proof.get(digest) {
```

Comment:
the difference between Some(Some(_)) is not very obvious to me vs Some(None), does it make more sense to get first then compare time?

```
if let Some(Some(proof)) = self.digest_proof.get(digest) {
  if *expiration >= current_time {
  } else {
  }
}
```